assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -2384,10 +2384,13 @@ public boolean canSubmit(Assignment assignment, String userId) {
                 // If an Extension exists for the user, we switch out the assignment's overall
                 // Close date for the extension deadline. We do this if the grade has been actually
                 // released, or if the submission object has not actually been submitted yet.
+                // Or when instructor granted an extension without returning grade/feedback
+                // and the user has created a submission that is ready to be submitted.
                 // Additionally, we make sure that a Resubmission date is not set [make sure it's null],
                 // so that this date-switching happens ONLY under Extension-related circumstances.
                 if (submission.getProperties().get(AssignmentConstants.ALLOW_EXTENSION_CLOSETIME) != null
-                        && (submission.getReturned() || !submission.getUserSubmission())) {
+                        && ((submission.getReturned() || !submission.getUserSubmission()) ||
+                            (!submission.getReturned() && submission.getUserSubmission()))) {
```

Comment:
this seems to always be true??

Reply:
I agree with Sam here, that this addition simply changes this if statement to

```if (submission.getProperties().get(AssignmentConstants.ALLOW_EXTENSION_CLOSETIME) != null) {```